vmm/src/vm_config.rs

```diff
@@ -115,7 +121,15 @@ pub struct MemoryZoneConfig {
     #[serde(default)]
     pub prefault: bool,
 }
-
+impl MemoryZoneConfig {
+    pub fn apply_landlock(self, landlock: &mut Landlock) -> LandlockResult<()> {
```

Comment:
The first parameter should be `&self`. There is no need to consume the structure.